### DISCUSSION TITLE
connecting API to planetscale db

### DISCUSSION THREAD

**Discussion by esaputras** [planetscale/discussion#591] [Q&A]
hello, I have a problem while connecting my API to my db that stored in planetscale.. I am using CodeIgniter 3 framework and I already follow the documentation here
![image1](https://github.com/planetscale/discussion/assets/19890746/89fa09d6-f10f-4555-8188-85b6efd21528)

after that, i got this error
![image2](https://github.com/planetscale/discussion/assets/19890746/82f9a008-5ea6-46fd-a00b-3b43e36763c9)

if you need some information, here is my config
![image3](https://github.com/planetscale/discussion/assets/19890746/0389d90c-fe9e-4417-b91e-1d2d01d52d07)

**Comment by orware** [comment]
Hello @esaputras,

I believe you have your configuration pretty much all set correctly although it may also be helpful to add an `'ssl_verify' => true,` entry in as well after the `ssl_ca` value.

The error you are receiving seems to be due to the following if statement returning true:
https://github.com/bcit-ci/CodeIgniter/blob/ae7b30f9cbaf462a3cc039de6d1e7738c6bea2d4/system/database/drivers/mysqli/mysqli_driver.php#L214-L218

There are valid reasons why this occurs for a PlanetScale database specifically, but since certain frameworks / CMS platforms use similar checks, the Joomla CMS is another example where a similar check fails, it can cause a false negative when connecting securely to a PlanetScale database.

In this situation I would recommend commenting out or removing lines 214-224 in `mysqli_driver.php` file above when using CodeIgniter 3 with a PlanetScale database and see if this helps to allow you to get past the error message you were experiencing last week.

If you continue to experience difficulty, you may also reach out via https://support.planetscale.com/ and we can work with you more closely if needed.

**Reply by esaputras** [reply]
I already solved it recently, the problem was on the tlsConfig. After I add the tls config "?tls=true&interpolateParams=true" on the db connection its work :)
